projects/simplicial-complex-operators/src/simplicial-complex-operators.cpp

```diff
@@ -170,8 +271,39 @@ bool SimplicialComplexOperators::isComplex(const MeshSubset& subset) const {
  */
 int SimplicialComplexOperators::isPureComplex(const MeshSubset& subset) const {
 
-    // TODO
-    return -1; // placeholder
+    if ( !isComplex(subset) ) return -1;
+
+    if (subset.vertices.size() == 0) return -1;
+
+    if (subset.edges.size() == 0) return 0;
+
+    if (subset.faces.size() == 0) return 1;
+
+    auto complexEdges = buildEdgeVector(subset);  
+    auto vertexToEdge = buildVertexEdgeAdjacencyMatrix();
+    auto vertices = vertexToEdge.transpose() * complexEdges;
+
+    for(Vertex vertex: mesh->vertices()) {
+        size_t i = vertex.getIndex();
+        if (subset.vertices.find(i) != subset.vertices.end() && vertices(i) == 0) {
```

Comment:
I did't really get what this condition for..
It means that if (subset.vertices.find(i) == subset.vertices.end()) or (vertices(i) != 0) than the vertex i suits us.. Why?